rpmlint/checks/PythonCheck.py

```diff
@@ -0,0 +1,76 @@
+from pathlib import Path
+import re
+
+from rpmlint.checks.AbstractCheck import AbstractFilesCheck
+
+# Warning messages
+WARNS = {
+    'doc': 'python-doc-in-package',
+}
+
+# Error messages
+ERRS = {
+    'egg-distutils': 'python-egg-info-distutils-style',
+    'tests': 'python-tests-in-site-packages',
+    'doc': 'python-doc-in-site-packages',
+    'src': 'python-src-in-site-packages',
+    'tests-package': 'python-tests-in-package',
+}
+
+SITELIB_RE = '/usr/lib[^/]*/python[^/]*/site-packages'
+
+# Paths that shouldn't be in any packages, ever, because they clobber global
+# name space.
+ERR_PATHS = [
+    (re.compile(f'{SITELIB_RE}/tests?$'), 'tests'),
+    (re.compile(f'{SITELIB_RE}/docs?$'), 'doc'),
+    (re.compile(f'{SITELIB_RE}/src$'), 'src'),
+    (re.compile(f'{SITELIB_RE}/[^/]+/tests?$'), 'tests-package'),
```

Comment:
So site-packages/foo/tests is an error, but site-packages/foo/bar/tests is fine?

```pycon
>>> import re
>>> SITELIB_RE = '/usr/lib[^/]*/python[^/]*/site-packages'
>>> r = re.compile(f'{SITELIB_RE}/[^/]+/tests?$')
>>> r.match('/usr/lib/python3.11/site-packages/foo/tests')
<re.Match object; span=(0, 43), match='/usr/lib/python3.11/site-packages/foo/tests'>
>>> r.match('/usr/lib/python3.11/site-packages/foo/bar/tests')
```

That does not sound like reasonable behavior to me.

Reply:
Good point. This is to detect a common pattern, that usually have tests in the package module, but it's true that tests can be found in any subpackage so I think that it shouldn't be difficult to find python packages that have this kind of folders/subfolders, so maybe in this case it's better to look for all the tests folders:

```python
    (re.compile(f'{SITELIB_RE}/.+/tests?$'), 'tests-package'),
```

Reply:
BTW this is my only remaining concern.

Reply:
This does not seem resolved.